src/libre/parser.act

```diff
@@ -77,6 +78,15 @@
 	#define TOK_CLASS__nspace  TOK_CLASS_NSPACE
 	#define TOK_CLASS__ndigit  TOK_CLASS_NDIGIT
 
+	/* This is a hack to work around the AST files not being able to include lexer.h. */
```

Comment:
Because that type, `struct ast_pos`, is defined in `re_char_class.h` and used in `re_char_class.c` (as well as the `.act` file), which is compiled once rather than per-dialect, so it doesn't have access to the dialect-specific details. The `lx_pos` struct it's duplicating is the same for every dialect, but that isn't known when building that .c file.